assets/src/js/admin/reports/components/no-data-notice/index.js

```diff
@@ -0,0 +1,77 @@
+// Dependencies
+import { useState, Fragment } from 'react';
+const { __ } = wp.i18n;
+
+// Store-related dependencies
+import { useStoreValue } from '../../store';
+
+// Styles
+import './style.scss';
+
+const NoDataNotice = ( { version } ) => {
+	// Get 'period' object from the store
+	const [ {}, dispatch ] = useStoreValue();
+
+	// Dispatch SET_RANGE action
+	const disablePeriodSelector = () => {
+		dispatch( {
+			type: 'ENABLE_PERIOD_SELECTOR',
+			payload: false,
+		} );
+	};
```

Comment:
Same as the last comment; store the action function with the reducer.

Reply:
Done, see ref: https://github.com/impress-org/givewp/commit/ebde416562fb1670bc9707453abc2e4656084e2e